src/test/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcherTest.java

```diff
@@ -106,6 +118,42 @@ public void test_getItems_incorrectAuthentication() throws IOException {
                     assertThat(error.getExtensions().get("statusCode")).isEqualTo(HttpStatus.UNAUTHORIZED.value());
                 })
                 .and()
-                .assertThatField(ITEM_DATA_JSON_PATH).isNull();
+                .assertThatField(GET_ITEMS_DATA_JSON_PATH).isNull();
+    }
+
+    @Test
+    public void getItemByItemId_normal() throws Throwable {
+        String itemId = "1";
+        ObjectNode variable = objectMapper.createObjectNode();
+        variable.put("itemId", itemId);
+        GraphQLResponse graphQLResponse = graphQLTestTemplate
+                .perform(GET_ITEM_BY_ITEM_ID_GRAPHQL_RESOURCE, variable);
+        assertThat(graphQLResponse).isNotNull();
+        assertThat(graphQLResponse.isOk()).isTrue();
+        graphQLResponse.assertThatNoErrorsArePresent()
+                .assertThatDataField().isNotNull()
+                .and()
+                .assertThatField(GET_ITEM_BY_ITEM_ID_DATA_JSON_PATH)
+                .as(ItemEntity.class)
+                .hasFieldOrPropertyWithValue("id", 1L);
+    }
+
+    @Test
+    public void getItemByItemId_NotFound() throws Throwable {
```

Comment:
A small change to keep the same style with other test names - change the name to `getItemByItemId_notFound`.